cmd/kube-rbac-proxy/app/options/proxyoptions.go

```diff
@@ -133,7 +133,7 @@ func (o *ProxyOptions) ApplyTo(krpInfo *server.KubeRBACProxyInfo, authInfo *serv
 		return fmt.Errorf("failed to setup transport for upstream: %w", err)
 	}
 
-	if configFileName := o.ConfigFileName; len(configFileName) > 0 {
+	if configFileName := o.AuthzConfigFileName; len(configFileName) > 0 {
 		krpInfo.Authorization, err = parseAuthorizationConfigFile(configFileName)
```

Comment:
It is not necessarily part of the issue, but I don't think that this is necessary: `configFileName := o.AuthzConfigFileName`. Maybe we could drop `configFileName` and just use `o.AuthzConfigFileName` in `parseAuthorizationConfigFile`.